src/Compilers/Core/Portable/Operations/OperationExtensions.Clone.cs

```diff
@@ -0,0 +1,477 @@
+            {
+                return new LocalFunctionStatement(operation.LocalFunctionSymbol, Visit(operation.Body), ((Operation)operation).SemanticModel, operation.Syntax, operation.Type, operation.ConstantValue);
+            }
+
+            public override IOperation VisitInterpolatedStringExpression(IInterpolatedStringExpression operation, object argument)
+            {
+                return new InterpolatedStringExpression(VisitArray(operation.Parts), ((Operation)operation).SemanticModel, operation.Syntax, operation.Type, operation.ConstantValue);
+            }
+
+            public override IOperation VisitInterpolatedStringText(IInterpolatedStringText operation, object argument)
+            {
+                return new InterpolatedStringText(Visit(operation.Text), ((Operation)operation).SemanticModel, operation.Syntax, operation.Type, operation.ConstantValue);
+            }
+
+            public override IOperation VisitInterpolation(IInterpolation operation, object argument)
+            {
+                return new Interpolation(Visit(operation.Expression), Visit(operation.Alignment), Visit(operation.FormatString), ((Operation)operation).SemanticModel, operation.Syntax, operation.Type, operation.ConstantValue);
+            }
+
+            public override IOperation VisitIsPatternExpression(IIsPatternExpression operation, object argument)
+            {
+                return new IsPatternExpression(Visit(operation.Expression), Visit(operation.Pattern), ((Operation)operation).SemanticModel, operation.Syntax, operation.Type, operation.ConstantValue);
+            }
+
+            public override IOperation VisitConstantPattern(IConstantPattern operation, object argument)
+            {
+                return new ConstantPattern(Visit(operation.Value), ((Operation)operation).SemanticModel, operation.Syntax, operation.Type, operation.ConstantValue);
+            }
+
+            public override IOperation VisitDeclarationPattern(IDeclarationPattern operation, object argument)
+            {
+                return new DeclarationPattern(operation.DeclaredSymbol, ((Operation)operation).SemanticModel, operation.Syntax, operation.Type, operation.ConstantValue);
+            }
+
+            public override IOperation VisitPatternCaseClause(IPatternCaseClause operation, object argument)
+            {
+                return new PatternCaseClause(operation.Label, Visit(operation.Pattern), Visit(operation.GuardExpression), ((Operation)operation).SemanticModel, operation.Syntax, operation.Type, operation.ConstantValue);
+            }
+
+            public override IOperation VisitTupleExpression(ITupleExpression operation, object argument)
+            {
+                return new TupleExpression(VisitArray(operation.Elements), ((Operation)operation).SemanticModel, operation.Syntax, operation.Type, operation.ConstantValue);
+            }
+        }
```

Comment:
Do we have to remember to update the Cloner every time new bound nodes or operations are added? 
If so, I would suggest a test that has the list of known operation types and breaks any time a new one is added. That canary test can have a checklist including "implement Clone".